sklearn/metrics/pairwise.py

```diff
@@ -1116,9 +1118,20 @@ def manhattan_distances(X, Y=None):
         Y.sum_duplicates()
         D = np.zeros((X.shape[0], Y.shape[0]))
         _sparse_manhattan(X.data, X.indices, X.indptr, Y.data, Y.indices, Y.indptr, D)
-        return D
-
-    return distance.cdist(X, Y, "cityblock")
+        return xp.asarray(D, device=device_)
+
+    # Note: the _convert_to_numpy functions are used to support the array api
+    # for manhattan distances for now, as the scipy cdist function for
+    # manhattan distance does not provide array api support.
+    # TODO: investigate with developing our own custom functionality for
+    #  computing the manhattan distance using the array api specification.
+    return xp.asarray(
+        distance.cdist(
+            _convert_to_numpy(X, xp=xp), _convert_to_numpy(Y, xp=xp), "cityblock"
+        ),
+        device=device_,
+        dtype=_max_precision_float_dtype(xp=xp, device=device_),
```

Comment:
I feel like if `distance.cdist(...).dtype == float32"`, then we shouldn't be converting it to `float64` here.